pkg/controllers/certificatestore_controller.go

```diff
@@ -110,8 +125,10 @@ func GetCertificatesMap() map[string][]*x509.Certificate {
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *CertificateStoreReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	pred := predicate.GenerationChangedPredicate{}
```

Comment:
If possible, do you think we can add a comment about the predicate discussion we had offline and why it's necessary/ how it works?